assets/msalv2.js

```diff
@@ -13,11 +13,11 @@ var aadOauth = (function () {
 
   // Initialise the myMSALObj for the given client, authority and scope
   function init(config) {
-    // TODO: Add support for other MSAL / B2C configuration
+    // TODO: Add support for other MSAL configuration
     var msalConfig = {
       auth: {
         clientId: config.clientId,
-        authority: "https://login.microsoftonline.com/" + config.tenant,
+        authority: config.isB2C ? "https://" + config.tenant + ".b2clogin.com/" + config.tenant + ".onmicrosoft.com/" + config.policy + "/" : "https://login.microsoftonline.com/" + config.tenant,
```

Comment:
This might be more readable if it used ${} interpolation since the expressions are now quite large.
i.e. something like (untested):
config.isB2C
? 'https://${config.tenant}.b2clogin.com/${config.tenant}.onmicrosoft.com/${config.policy}/' 
: 'https://login.microsoftonline.com/${config.tenant}'

Reply:
Note: I'm also assuming that you've tested this as I stated previously, I wasn't sure what else was required to support B2C as I don't use it.

Reply:
I updated the PR with the requested change and tested that it launches the user flow correctly.

I'm still having problems getting the token, because when the user flow completes and returns a token to the app, aad_oauth does not save Azure's auth response and will re-initiate a login when calling login() again instead of returning the access token.

This problem is unrelated to the change above though, still debugging why this occurs.

Reply:
Which login flow are you using? Pop up or redirect? Are you able to get a token at all after the second attempt?

Reply:
Sorry, I was thinking this was dart code for some reason.  I don't think that change will work as I stated in the Javascript unless you use backtick quotes (`). It might be safer to just leave it as string concatenation.

Reply:
I'm not sure how B2C works with login accounts but it might be related to the simplistic getAccount() javascript method. It expects to only find one account in the myMSALObj and will return the first one if there's more than one.

If it can't find an account it will retry interactive login, since it needs the account to do the silent login (around line 90). If the silent acquireTokenSilent is throwing the error should be getting logged to the browser console.

Reply:
You are right, that type of string interpolation doesn't work. I didn't notice during testing because when I changed my package ref to the PR, it didn't automatically pull the latest update.

Change is reverted and code is back at the original version.

I'm still trying to debug the JavaScript code. Either the redirect handler or silent token retrieval fails. Hope to know soon.